File: frontend/src/dtos/Venta/create/VentaCreateRequestDto.ts
import DetalleVentaCreateRequestDto from '../../../dtos/DetalleVenta/create/DetalleVentaCreateRequestDto';
export default class VentaCreateRequestDto
{
	fecha : Date=new Date();
	numeroVenta : string='';
	detalles : DetalleVentaCreateRequestDto[] = [];

	setFecha : (fecha : Date) => VentaCreateRequestDto = (fecha: Date) => {
		this.fecha = fecha;
		return this;
	}

	setNumeroVenta : (numeroVenta : string) => VentaCreateRequestDto = (numeroVenta: string) => {
		this.numeroVenta = numeroVenta;
		return this;
	}

	addDetalles : () => VentaCreateRequestDto = () => {
		this.detalles.push(new DetalleVentaCreateRequestDto());
		return this;
	}

	deleteDetalles : (detalleVenta : DetalleVentaCreateRequestDto) => VentaCreateRequestDto = (detalleVenta : DetalleVentaCreateRequestDto) => {
		this.detalles = this.detalles.filter(x => x !== detalleVenta);
		return this;
	}
}

File: backend/src/main/java/com/example/template/dtos/VentaCreateRequestDTO.java
package com.example.template.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.*;
import com.example.template.enums.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class VentaCreateRequestDTO {
private Date Fecha;
private String NumeroVenta;
private List<DetalleVentaCreateRequestDTO>Detalles = new ArrayList<>();
}
